pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -215,9 +267,91 @@ def compute_derivatives(
             self.tracker.record()
 
         if execution_config.gradient_method == "adjoint":
-            res = tuple(adjoint_jacobian(circuit) for circuit in circuits)
+            max_workers = self._get_max_workers(execution_config)
+            if max_workers is None:
+                res = tuple(adjoint_jacobian(circuit) for circuit in circuits)
+            else:
+                vanilla_circuits = [convert_to_numpy_parameters(c) for c in circuits]
+                with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
+                    exec_map = executor.map(adjoint_jacobian, vanilla_circuits)
+                    res = tuple(circuit for circuit in exec_map)
+
+                # reset _rng to mimic serial behavior
+                self._rng = np.random.default_rng(self._rng.integers(2**31 - 1))
+
             return res[0] if is_single_circuit else res
 
         raise NotImplementedError(
             f"{self.name} cannot compute derivatives via {execution_config.gradient_method}"
         )
+
+    # pylint: disable=missing-function-docstring
+    def _get_max_workers(self, execution_config=None):
+        max_workers = None
+        if execution_config is not None:
+            if isinstance(execution_config.device_options, dict):
+                if "max_workers" in execution_config.device_options.keys():
+                    max_workers = execution_config.device_options["max_workers"]
+        if max_workers is None:
+            max_workers = self._max_workers
+        _validate_multiprocessing_workers(max_workers)
+        return max_workers
+
+    def _validate_multiprocessing_circuits(self, circuits):
+        """Make sure the tapes can be processed by a ProcessPoolExecutor instance.
+
+        Args:
+            circuits (QuantumTape_or_Batch): Quantum tapes
+        """
+        if self._debugger is not None:  # pragma: no cover
+            if self._debugger.active:
+                raise DeviceError(
+                    "Debugging with ``Snapshots`` is not available with multiprocessing."
+                )
```

Comment:
```suggestion
        if self._debugger and self._debugger.active:  # pragma: no cover
            raise DeviceError(
                "Debugging with ``Snapshots`` is not available with multiprocessing."
            )
```

Reply:
> I saw a test for this line

Do you mean `test_snapshot_multiprocessing`? For me the lines below got covered, but maybe we don't want to necessarily forbid Snapshot's, but just `_debugger.active`?

Reply:
Never mind, I added the test you suggested for the debugger.